Source/EmbeddedBoundary/ParticleScraper.H

```diff
@@ -156,10 +156,14 @@ scrapeParticles (PC& pc, const amrex::Vector<const amrex::MultiFab*>& distance_t
             auto& tile = pti.GetParticleTile();
             auto ptd = tile.getParticleTileData();
             const auto np = tile.numParticles();
+            auto& particles = tile.GetArrayOfStructs();
             auto phi = (*distance_to_eb[lev])[pti].array();  // signed distance function
             amrex::ParallelForRNG( np,
             [=] AMREX_GPU_DEVICE (const int ip, amrex::RandomEngine const& engine) noexcept
             {
+                // skip particles that are already flagged for removal
+                if (particles[ip].id() < 0) return;
```

Comment:
Others can chime in, but I think using `return` in the lambda expression is not a good idea. On a CPU, the `ParallelFor` gets expressed as a loop, so the `return` returns out of the whole routine.

Reply:
I've seen `return` used in other `ParallelFor` in the code (see below for example), so I guess it should be ok (also, the CPU tests have all passed).

https://github.com/ECP-WarpX/WarpX/blob/5a5f8f89ecd4294d271bb66aa0e236887243ba67/Source/Particles/WarpXParticleContainer.cpp#L1052-L1059

Reply:
`return` will return from the lambda function call for that loop element, rather than the enclosing routine. Using `continue` in this context, however, which is easy to do when porting a regular C++ `for` loop to an `amrex::ParallelFor`, is usually a bug because it doesn't actually break out of the lambda function.

Reply:
Ok, thanks. I remember reading about the problem with returns in the past. Did this change at some point?

Reply:
Not for `ParallelFor`, but `return` was and still is problematic for macros like `AMREX_FOR_1D`.